src/buffer.rs

```diff
@@ -229,20 +253,38 @@ impl Buffer {
 		}
 	}
 
-	pub fn insert_at_point<'b>(&mut self, point: Point, text: &'b str) -> Point {
+	/// Inserts text in the buffer at the provided point
+	pub fn insert_at_point<'b>(&mut self, point: Point, text: String) -> Point {
 		let index = self.content.line_to_byte(point.row) + point.column;
-		let mut point = self.edit_region(index, index, text);
+		let mut point = self.edit_region(index, index, text.as_str());
+		// ensure that we get the correct text position after the insert
 		point.column += text.len();
```

Comment:
we need an advance/previous function soon to handle wrapping, i think

Reply:
i thought wrapping would be handled by the rendering and not be stored internally in the text?

Reply:
position of point needs to be wrapped, i mean
point.column at 5 + 5 in a 6 column line should increment row and set column modulo, for arbitrary column advance (i guess it degrades to arbitrary char point lookup in large steps)